src/Swagger/DocumentationNormalizer.php

```diff
@@ -208,6 +176,33 @@ public function normalize($object, $format = null, array $context = [])
         return $doc;
     }
 
+    private function getPath(string $resourceClass, ResourceMetadata $resourceMetadata, string $operationName, bool $collection)
+    {
+        // Check if the operation has a custom path
+        $method = $collection ? 'getCollectionOperationAttribute' : 'getItemOperationAttribute';
+        $path = $resourceMetadata->{$method}($operationName, 'path');
+        if (null !== $path) {
+            return $path;
+        }
+
+        if (null === $this->iriConverter) {
+            throw new InvalidArgumentException(sprintf('%s can\'t manage the %s %s operation of the resource "%s" because it needs an instance of %s to manage standard operations (path not defined manually).', __CLASS__, $collection ? 'collection' : 'item', $resourceClass, IriConverterInterface::class));
+        }
+
+        // @todo Fix the limitation of the IriConverter when a resource
+        // has no collection operations
```

Comment:
There were several reasons:

1. The class is useless: only the key matter, the value (the context array) is implementation specific. At time of removing there was no key always available.
2. The context carries no semantic, it can contain routing data, Hydra context of anything used by any piece or extension of API Platofrm. An associative array is more adapted than a class in this case.
3. It makes it more easy to write annotations in the 
Entity, and more readable.

Why do you need/want a class? To store the URL of the operation? It can maybe make sense, I've to take a deeper look.